### PR TITLE
Add text field default value field to settings model and display it in settings edit view

### PR DESCRIPTION
**Pull request by BinderDyn**
#12547

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=12548) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=12548) before we can accept your contribution.<br/><br/>:x: BinderDyn [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=12548)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=12548) it.</sub>

**Comment by Skrypt** [comment]
The default value should be optional. The reason is that this field is used by many already without a Default value. You need to make it backward compatible. What happens if I have a TextField that doesn't have a DefaultValue setting in its content definition?

**Comment by BinderDyn** [comment]
Didn't think if that but you're right. I'll rework the existing solution taking that into account.

**Comment by BinderDyn** [comment]
@Skrypt I tested this with the given scenario. It is optional and does not break existing text fields. If the property of the settings was not present it simply returns null for DefaultValue. Because it is a string no harm is done there. It simply stays empty in the UI as there was no default set.

**Comment by sebastienros** [comment]
Please check the issue comments, at least this PR implements the easy case.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDisplayDriver.cs [hunk 1]
```diff
@@ -38,7 +38,8 @@ public override IDisplayResult Edit(TextField field, BuildFieldEditorContext con
         {
             return Initialize<EditTextFieldViewModel>(GetEditorShapeType(context), model =>
             {
-                model.Text = field.Text;
+                var settings = context.PartFieldDefinition.GetSettings<TextFieldSettings>();
+                model.Text = context.IsNew ? settings.DefaultValue : field.Text;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDisplayDriver.cs:42]: If we were to take this PR,  you may want to change this line to 

model.Text = context.IsNew && field.Text == null ? settings.DefaultValue : field.Text;

Comment (BinderDyn) [reply src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDisplayDriver.cs]: Implemented with newest commit

**Comment by MikeAlhayek** [comment]
@sebastienros I think this is a fair ask since we already do similar thing in couple other field drivers.

**Review by MikeAlhayek — APPROVED** [review]
@sebastienros I am approving this as we already do the same thing with other fields. I think it is a fair ask. However, if you don't want this merged please close it.

**Review by MikeAlhayek — APPROVED** [review]
@sebastienros I am approving this as we already do the same thing with other fields. I think it is a fair ask. However, if you don't want this merged please close it.

**Comment by sebastienros** [comment]
Approved since we already discussed it with you @MikeAlhayek recently for other fields and I believe the same solution is applied here. With a reminder that this won't work for anything that is not driven by the UI (imports, api calls, ...)

**Comment by sebastienros** [comment]
A solution to the limitation might be to be more explicit in the labels/hints, like "will be used to pre-populate the field editor when the item is in creation"

**Pull request merged**